admin/coupon_admin.php

```diff
@@ -1408,7 +1408,11 @@ function check_form(form_name) {
       $contents[] = array('text' => '<br />' . COUPON_USES_COUPON . '<br />' . zen_draw_input_field('voucher_number_of'));
       break;
     default:
-      $heading[] = array('text'=>'['.$cInfo->coupon_id.']  '.$cInfo->coupon_code . ($cInfo->coupon_id == '' ? ' - (' . (!empty($_GET['cid']) ? $_GET['cid'] : 0) . ')' : ''));
+      if (!$cc_list->EOF) { 

```

Comment:
Shouldn't this check against `$cc_list->RecordCount() != 0` or some other comparison instead?

If followed the code correctly, when there are items in `$cc_list` and are processing through the `default:` action, that the list is stepped through using `while (!$cc_list->EOF) { … $cc_list->MoveNext(); }` and therefore after going through all of the items in the list that `$cc_list->EOF === true`...  I haven't run it, but that's what it looks like on the surface...

Reply:
Good catch!  TY.